### PR TITLE
part-of-Mukund Sivaraman review

### PR DESCRIPTION
**Pull request by boucadair**

### CODE REVIEW

File: draft-ietf-dnsop-structured-dns-error.md [hunk 1]
```diff
@@ -236,21 +236,16 @@ additional information about the cause of a DNS error.
...
 endpoint from malicious software, "Phishing" to prevent the user from
-revealing sensitive information to the attacker, etc. A user needs to
+revealing sensitive information to the attacker, etc. A user may need to
 know the contact details of the IT/InfoSec team to raise a complaint.
 
-5. When a DNS resolver or forwarder forwards the received EDE option, the
-EXTRA-TEXT field only conveys the source of the error ({{Section 3 of
-!RFC8914}}) and does not provide additional textual information about
-the cause of the error.
-
```
Comment (boucadair) [inline draft-ietf-dnsop-structured-dns-error.md:250]: The current wording contradicts this part from the EDE spec:

   When a resolver or forwarder receives an EDE option, whether or not
   (and how) to pass along EDE information on to their original client
   is implementation dependent.  Implementations MAY choose to not
   forward information, or they MAY choose to create a new EDE option(s)
   that conveys the information encoded in the received EDE.  When doing
   so, the source of the error SHOULD be attributed in the EXTRA-TEXT
   field, since an EDNS0 option received by the original client will
   appear to have come from the resolver or forwarder sending it.

The spec does explain clearly how this will be "attributed" though. That's said, we do already have a fair list of motivations, no need to be exhaustive here.

File: draft-ietf-dnsop-structured-dns-error.md [hunk 2]
```diff
@@ -84,10 +84,10 @@ informative:
 --- abstract
 
 DNS filtering is widely deployed for various reasons, including
-network security. However, filtered DNS responses lack information
+network security. However, filtered DNS responses lack structured information
 for end users to understand the reason for the filtering.
 Existing mechanisms to provide explanatory details to end users cause harm
-especially if the blocked DNS response is to an HTTPS server.
+especially if the blocked DNS response is for HTTPS resources.
```
Comment (tireddy2) [inline draft-ietf-dnsop-structured-dns-error.md:90]: What is wrong with the previous line using "HTTPS server" ?

File: draft-ietf-dnsop-structured-dns-error.md (hunk 1 above)
Comment (tireddy2) [reply draft-ietf-dnsop-structured-dns-error.md:250]: We don't have to delete the line, it can be fixed as follows:
When a DNS resolver or forwarder forwards the received EDE option, the
EXTRA-TEXT field can convey the source of the error ({{Section 3 of
!RFC8914}}) and does not provide additional textual information about
the cause of the error.

Comment (boucadair) [reply draft-ietf-dnsop-structured-dns-error.md:250]: The point of the reviewer is that the EDE is an information about the cause itself. Which I think is OK. The key contrib of the spec is to structure the info and provide more details (c, j, etc.). 

This bullet does not bring much, IMO

File: draft-ietf-dnsop-structured-dns-error.md (hunk 2 above)
Comment (boucadair) [reply draft-ietf-dnsop-structured-dns-error.md:90]: This is to address this comment: 

> "I suggest rewording this sentence as it reads as though the blocked DNS response is sent to a HTTPS server. Instead, I assume what the authors are attempting to state is that blocked DNS responses may cause harm if they contain address records that cause web traffic for a web-domain to be directed to a different local HTTPS server."
